slides-battle-react/src/components/battle/Battle.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect } from "react";
+import { Button, Card } from "@material-ui/core";
+import { BattleHeader } from "./BattleHeader";
+import {
+  PlayCircleOutline,
+  HighlightOff,
+  NavigateNext,
+} from "@material-ui/icons";
+import { Header } from "../layout/Header";
+
+export const Battle = ({
+  isBattleEnabled,
+  isBattlePlaying,
+  startBattle,
+  stopBattle,
+  round,
+  startRound,
+  availablePlayers,
+  switchConfiguration,
+}) => {
+  useEffect(() => {
+    if (isBattlePlaying && !round.id) {
+      startRound();
+    }
+  }, [startRound, isBattlePlaying, round]);
+
+  return (
+    <>
+      <BattleHeader backgroundColor="#433826" onClick={switchConfiguration} />
+      <div
+        style={{

```

Comment:
How the style rules will be generated in the built application : inline, separate file, etc ... ?

Reply:
I am not a big fan of external css files (styled components is there for that)
For static styles, we can either use inline or styled component
In my experience
- I use styled components when the style may be dynamic and/or reused in other components 
- I use inline when I just need to handle some specific style in the component rendering

But, I can remove the inline style to use styled components everywhere

I am not a big fan of tons of css files, we had a lot of problems with classNames in previous big projects (~ 80k lines) because when you reuse components, you may not be aware of an existing style on the current className, or someone who overrides your own style, etc ..